openpype/modules/clockify/clockify_api.py

```diff
@@ -236,231 +217,219 @@ def get_tag_id(self, tag_name, workspace_id=None):
             return None
         return all_tasks[tag_name]
 
-    def get_task_id(
-        self, task_name, project_id, workspace_id=None
-    ):
+    def get_task_id(self, task_name, project_id, workspace_id=None):
         if workspace_id is None:
             workspace_id = self.workspace_id
-        all_tasks = self.get_tasks(
-            project_id, workspace_id
-        )
+        all_tasks = self.get_tasks(project_id, workspace_id)
         if task_name not in all_tasks:
             return None
         return all_tasks[task_name]
 
     def get_current_time(self):
-        return str(datetime.datetime.utcnow().isoformat())+'Z'
+        return str(datetime.datetime.utcnow().isoformat()) + "Z"
 
     def start_time_entry(
-        self, description, project_id, task_id=None, tag_ids=[],
-        workspace_id=None, billable=True
+        self,
+        description,
+        project_id,
+        task_id=None,
+        tag_ids=None,
+        workspace_id=None,
+        user_id=None,
+        billable=True,
     ):
         # Workspace
         if workspace_id is None:
             workspace_id = self.workspace_id
+        # User ID
+        if user_id is None:
+            user_id = self._user_id
+
+        # get running timer to check if we need to start it
+        current_timer = self.get_in_progress()
+        current_timer_hierarchy = None
+        current_project_id = None
+        current_task_id = None
 
         # Check if is currently run another times and has same values
-        current = self.get_in_progress(workspace_id)
-        if current is not None:
+        # DO not restart the timer, if it is already running for curent task
+        if current_timer and current_timer is not None:
+            current_timer_hierarchy = current_timer.get("description")
+            current_project_id = current_timer.get("projectId")
+            current_task_id = current_timer.get("taskId")
             if (
-                current.get("description", None) == description and
-                current.get("projectId", None) == project_id and
-                current.get("taskId", None) == task_id
+                current_timer
+                and current_timer is not None
+                and description == current_timer_hierarchy
```

Comment:
It would not get to this point if current timer would not be set.
```suggestion
                description == current_timer_hierarchy
```